### PR TITLE
Fixed bugs in the optimization doc.

### PR DESCRIPTION
**Pull request by mbaudin47**
* Fixed issues in the `OptimizationProblem` class doc.
* Fixed issues in the `OptimizationResult` class doc.
* Created 2 new examples for Rosenbrock and Rastrigin test functions. 
* Improved the EGO examples. Fixed the equations. Changed the design of experiments from factorial designs to LHS experiments. Added graphics as suggested in #1302. However, this is not sufficient to make them work: they never worked after all! This might be related to #1432.

### CODE REVIEW

File: python/src/NearestPointProblem_doc.i.in [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+    \min_{x} \frac{1}{2}\|x\|_2^2 \\
     g(x) = v
 
+where :math:`\| \cdo t\|_2` is the euclidian norm. 
+
+In other words, this is a minimum norm problem 
+with a general nonlinear constraint. 
+
+Available constructors:
+    NearestPointProblem(*level, value*)
```
Comment (jschueller) [inline python/src/NearestPointProblem_doc.i.in:17]: Dont list the ctors when there's only one available.

Comment (mbaudin47) [reply python/src/NearestPointProblem_doc.i.in:17]: I see the reasoning, but I think that any class should have its constructors documented: doing so enables to just copy/paste the constructor into the code. Are there other classes with the same property?

Comment (jschueller) [reply python/src/NearestPointProblem_doc.i.in:17]: Yes, a lot!

Comment (mbaudin47) [reply python/src/NearestPointProblem_doc.i.in:17]: Ok: will fix this.

**Comment by jschueller** [comment]
- setImprovementFactor defaults to 0, so you can mention that it's not mandatory to set it to 0
- the example that depend on NLopt must be disabled in python/test/CmakeLists when nlopt is unavailable, look for WILL_FAIL keyword
- NLopt is newer written NLOPT

**Comment by mbaudin47** [comment]
>     * setImprovementFactor defaults to 0, so you can mention that it's not mandatory to set it to 0

I did not notice the very recent change dcb4ca989c85b4bad6e1c03d8c003fc4c94762c0 which, indeed, makes this as the default. I will fix this, but I need to re-build OT so that I can generate the updated NB.

File: python/src/NearestPointProblem_doc.i.in (hunk 1 above)
Comment (mbaudin47) [reply python/src/NearestPointProblem_doc.i.in:17]: Done at d9d32b2de6295f4d6f51fc583c7fdeff032cbfaf

**Comment by mbaudin47** [comment]
>     * setImprovementFactor defaults to 0, so you can mention that it's not mandatory to set it to 0
> 
>     * the example that depend on NLopt must be disabled in python/test/CmakeLists when nlopt is unavailable, look for WILL_FAIL keyword
> 
>     * NLopt is newer written NLOPT

Fixed at 86d35dc84a5d07316e2c70db8fa84d440d48a057 and 68f59d0fe534bed82d4cbc5b8d61675c8c61e615

**Pull request merged**